timeseries/src/series.rs

```diff
@@ -0,0 +1,319 @@
+    Gauge,
+    /// A monotonically increasing metric.
+    Counter,
+}
+
+/// A time series with its identifying labels and data points.
+///
+/// A series represents a single stream of timestamped values.
+///
+/// # Identity and Metadata
+///
+/// A series is uniquely identified by its `name` and `labels`. The `metric_type`,
+/// `unit`, and `description` fields are metadata with last-write-wins semantics.
+///
+/// # Example
+///
+/// ```
+/// use timeseries::{Series, Label, Sample};
+///
+/// let series = Series::new(
+///     "http_requests_total",
+///     vec![Label::new("method", "GET")],
+///     vec![Sample::new(1700000000000, 1.0)],
+/// );
+///
+/// // Or use the builder:
+/// let series = Series::builder("http_requests_total")
+///     .label("method", "GET")
+///     .sample(1700000000000, 1.0)
+///     .build();
+/// ```
+#[derive(Debug, Clone)]
+pub struct Series {
+    // --- Identity ---
+    /// The metric name.
+    pub name: String,
+
+    /// Labels identifying this series.
+    pub labels: Vec<Label>,
+
+    // --- Metadata (last-write-wins) ---
+    /// The type of metric (gauge or counter).
+    pub metric_type: Option<MetricType>,
+
+    /// Unit of measurement (e.g., "bytes", "seconds").
+    pub unit: Option<String>,
+
+    /// Human-readable description of the metric.
+    pub description: Option<String>,
+
+    // --- Data ---
+    /// One or more samples to write.
+    pub samples: Vec<Sample>,
+}
+
+impl Series {
+    /// Creates a new series with the given name, labels, and samples.
+    pub fn new(name: impl Into<String>, labels: Vec<Label>, samples: Vec<Sample>) -> Self {
+        Self {
+            name: name.into(),
+            labels,
+            metric_type: None,
+            unit: None,
+            description: None,
+            samples,
+        }
+    }
+
+    /// Creates a builder for constructing a series.
+    ///
+    /// The builder provides a fluent API for creating series with
+    /// labels, samples, and metadata fields.
+    ///
+    /// # Arguments
+    ///
+    /// * `name` - The metric name.
+    pub fn builder(name: impl Into<String>) -> SeriesBuilder {
+        SeriesBuilder::new(name)
+    }
+}
+
+/// Builder for constructing [`Series`] instances.
+///
+/// Provides a fluent API for creating series with labels, samples,
+/// and metadata fields.
+#[derive(Debug, Clone)]
+pub struct SeriesBuilder {
+    name: String,
+    labels: Vec<Label>,
+    metric_type: Option<MetricType>,
+    unit: Option<String>,
+    description: Option<String>,
+    samples: Vec<Sample>,
+}
+
+impl SeriesBuilder {
+    fn new(name: impl Into<String>) -> Self {
+        Self {
+            name: name.into(),
+            labels: Vec::new(),
+            metric_type: None,
+            unit: None,
+            description: None,
+            samples: Vec::new(),
+        }
+    }
+
+    /// Adds a label to the series.
+    pub fn label(mut self, name: impl Into<String>, value: impl Into<String>) -> Self {
```

Comment:
consider `add_label` and `add_sample` to make it clear that you can add multiple (as opposed to something like `unit` which overwrites)